worker-document/src/main/java/com/hpe/caf/worker/document/scripting/specs/AbstractScriptSpec.java

```diff
@@ -51,14 +53,21 @@ public static AbstractScriptSpec create(final ApplicationImpl application, final
             throw new InvalidScriptException(script, "Script must have a single source.");
         }
 
+        final ScriptEngineType engineType;
+        if (script.engine == null || script.engine.isEmpty()) {
+            engineType = ScriptEngineType.NASHORN;
+        } else {
+            engineType = ScriptEngineType.valueOf(script.engine);
```

Comment:
You haven't validated the engine yet so you don't know that this won't result in an `IllegalArgumentException`.  That is an unchecked exception that should only be happening when there is a fault in the code.  Although catching it here and re-throwing an `InvalidScriptException` would be fine rather than introducing earlier validation code.